app/middlewares/checkAuthorization.js

```diff
@@ -0,0 +1,19 @@
+const constants = require('../../lib/constants');
+const errors = require('../errors');
+
+const isOwner = (currentUserId, user) => parseInt(currentUserId) === user.id;
+
+const isAdmin = user => user.type === constants.user_types.ADMIN;
+
+module.exports.ownerAdmin = async (req, _, next) => {
+  const user = await req.currentUser;
+  try {
+    if ((isAdmin(user) || isOwner(req.params.userId, user)) === false) {
+      throw errors.invalidCredentials();
```

Comment:
Just a fyi, because I'd normally ask you to have different descriptions for each error.
Since this is a security validation, we often don't want to tell people why their query failed, since that ends up giving them more information to attack the system, so in this case the lack of information is ok.